basics_of_logging_in_sst-core/Simplified_SST_Logging.md

```diff
@@ -0,0 +1,329 @@
+
+### Writing Log Messages
+
+The SST Output class provides four different methods to write log messages.
+
+**Output.**  This method works like a standard printf.  By default its messages do not contain the prefix data but can if desired.  
+
+**Verbose.**  Messages logged using this method contain the prefix method by default.  Whether or not the messages are logged is controlled by the verbose level and verbose mask.
+
+**Debug.**  This method operates similarly to the Verbose method but will only be compiled if the --enable-debug-flag is set in the SST configuration file.  
+
+**Fatal.**  Messages logged using this method are always sent to SST::Output::STDERR and terminate the simulation.  
+
+Of the four methods, Verbose is the one that will be used most often.
+
+### SST Logging Example
+
+Based upon the above information, a typical logger instantiation will look like:
+
+```
+Output logger("@t; File=@f; Func=@p; Line=@l", 300, 0x00, SST::Output::STDOUT);
+```
+This logger will include the simulation time, file name, function name, and line number in the prefix.  Any log message with a verbose level of less than 300 and a verbose mask of 0x00 will print to STDOUT.
+
+Calls to write to the log file will look like:
+
+```
+logger.verbose(CALL_INFO, 200, 0x00, "Log message %d.\n", 1);
+logger.verbose(CALL_INFO, 400, 0x00, "Log message %d.\n", 2);
+```
+In these cases, the first log message with a verbose level less than 300 will be printed.  However, the second will not be printed since its verbose level is greater than 300.
+
+## SIMPLIFYING THE LOGGER PROCESS
+
+The SST Output class provides all the capabilities necessary for logging.  However, there are some ways the interface can be made simpler and more intuitive.
+
+### Define Log Levels ###
+
+The SST Output class defines its verbose level as a 32-bit unsigned integer.  This allows for over 4 billion logging levels.  However, loggers typically only define a few logging levels as listed above, possibly with room for a few user-defined custom logging levels.  In addition, loggers typically define their logging levels by name for easy reference.
+
+To accomplish this in SST, map the numeric log levels to names using `#define` or enumerations.
+
+```
+#define LOG_LEVEL_FATAL 0
+#define LOG_LEVEL_ERROR 100
+#define LOG_LEVEL_WARN  200
+#define LOG_LEVEL_INFO  300
+#define LOG_LEVEL_DEBUG 400
+#define LOG_LEVEL_TRACE 500
+
+enum LogLevel {
+    Fatal = 0,
+    Error = 100,
+    Warn  = 200,
+    Info  = 300,
+    Debug = 400,
+    Trace = 500
+}
+```
+
+Once the log levels are defined instantiation and calling the logger will look like:
+
+```
+Output logger("@t; File=@f; Func=@p; Line=@l", INFO, 0x00, SST::Output::STDOUT);
```

Comment:
In general, you'll want to pass the level into the Component as a parameter, not set it statically in the constructor.  This also means you'll have to have some way to map the string versions of the supported levels to the proper integer, or have the end user pass the verbose level in as an integer (which then somewhat defeats the use of #define or enums).  I think this is an important point that should be brought out in all or some of the more advanced examples.